src/mapproxyUtils/mapproxySeed.ts

```diff
@@ -0,0 +1,264 @@
+import { promises as fsp } from 'node:fs';
+import { $ } from 'zx';
+import { dump } from 'js-yaml';
+import { Logger } from '@map-colonies/js-logger';
+import { inject, singleton } from 'tsyringe';
+import { SERVICES } from '../common/constants';
+import { IConfig, ISeedBase, ITaskSeedOptions } from '../common/interfaces';
+import { MapproxyConfigClient } from '../clients/mapproxyConfig';
+/* eslint-disable @typescript-eslint/naming-convention */
+import { BaseCache, Cleanup, Seed, seedsSchema, cleanupsSchema, baseSchema } from '../common/schemas/seeds';
+import { Coverage, coveragesSchema } from '../common/schemas/coverages';
+import { SeedMode } from '../common/enums';
+import { fileExists, isValidDateFormat, zoomComparison } from '../common/validations';
+
+$.verbose = false;
+
+@singleton()
+export class MapproxySeed {
+  private readonly mapproxyYamlDir: string;
+  private readonly seedYamlDir: string;
+  private readonly geometryCoverageFilePath: string;
+  private readonly seedConcurrency: number;
+  private readonly mapproxySeedProgressDir: string;
+
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.CONFIG) private readonly config: IConfig,
+    private readonly mapproxyConfigClient: MapproxyConfigClient
+  ) {
+    this.mapproxyYamlDir = this.config.get<string>('mapproxy.mapproxyYamlDir');
+    this.seedYamlDir = this.config.get<string>('mapproxy.seedYamlDir');
+    this.geometryCoverageFilePath = this.config.get<string>('mapproxy.geometryTxtFile');
+    this.seedConcurrency = 5;
+    this.mapproxySeedProgressDir = this.config.get<string>('mapproxy.seedProgressFileDir');
+  }
+
+  public async runSeed(task: ITaskSeedOptions): Promise<void> {
+    this.logger.info({
+      msg: `processing seed for job of ${task.layerId}`,
+      layerId: task.layerId,
+      grid: task.grid,
+      mode: task.mode,
+      fromZoom: task.fromZoomLevel,
+      toZoom: task.toZoomLevel,
+      refreshBefore: task.refreshBefore,
+    });
+
+    try {
+      await this.writeMapproxyYaml();
+      await this.writeGeojsonTxtFile(this.geometryCoverageFilePath, JSON.stringify(task.geometry));
+      await this.createSeedYamlFile(task);
+      await this.executeSeed(task);
+      this.logger.info(`complete seed for job of ${task.layerId}`);
+    } catch (err) {
+      this.logger.error({ msg: `failed seed for job of ${task.layerId}`, err });
+      throw new Error(`failed seed for job of ${task.layerId} with reason: ${(err as Error).message}`);
+    }
+  }
+
+  public async runClean(task: ITaskSeedOptions): Promise<void> {
```

Comment:
this method is duplication of "runSeed" method, please consider to unite the methods and to add the SEED options to the log